src/wms/src/main/java/org/geoserver/wms/capabilities/Capabilities_1_3_0_Transformer.java

```diff
@@ -703,8 +705,8 @@ private void handleLayers() {
             // handle root bounding box
             List<LayerGroupInfo> layerGroups;
             List<LayerInfo> layers;
-            SortBy lgOrder = asc("name");
-            SortBy order = asc("name");
+            SortBy lgOrder = asc("sortIndex");
```

Comment:
Not backwards compatible, the current sorting order is the desired default and must not be modified. Please see comment above. This case is complicated though, the catalog machinery does not support sorting on multiple SortBy conditions, it must be extended... :-(